app/controllers/school/base_controller.rb

```diff
@@ -12,12 +12,12 @@ def require_school_user!
   end
 
   def set_school
-    @school = @user.school
+    @school = @user.schools.size == 1 ? @user.school : School.find_by_urn(params[:urn])
```

Comment:
There's a couple of issues here:

1. Because you're not scoping to the user's schools, once a user has multiple schools, they can access any school they like.
2. A much lesser problem, more like unexpected behaviour: if the user has only one school, any URN they provide yields the school they're attached to, so it looks like a mismatch between the URL entered and the data returned.

The way to fix both:

```suggestion
    @school = @user.schools.find_by!(urn: params[:urn])
```

Reply:
I'm guessing we're missing a controller spec for the case:

- given a user with multiple schools
- when they to access a school they don't have access to
- then they get a 404 when